tests/contract.rs

```diff
@@ -153,3 +136,49 @@ fn add_file(cache: &mut FileResolver, path: &Path, target: Target) -> io::Result
 
     Ok(filename.to_string())
 }
+
+fn check_diagnostics(path: &Path, ns: &Namespace) {
+    let mut expected = "// ----\n".to_owned();
+
+    for diag in ns.diagnostics.iter() {
+        if diag.level == Level::Warning || diag.level == Level::Error {
+            expected = format!(
```

Comment:
Shouldn't this be `expected.push`? This looks like you are reassigning the variable.